File: educational-core/src/com/jetbrains/edu/learning/courseFormat/tasks/ChoiceTask.java
package com.jetbrains.edu.learning.courseFormat.tasks;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class ChoiceTask extends Task {

  @SuppressWarnings("unused") //used for deserialization
  public ChoiceTask() {}

  private List<String> myChoiceVariants = new ArrayList<>();
  private boolean myIsMultipleChoice;
  private List<Integer> mySelectedVariants = new ArrayList<>();

  public List<Integer> getSelectedVariants() {
    return mySelectedVariants;
  }

  public void setSelectedVariants(List<Integer> selectedVariants) {
    mySelectedVariants = selectedVariants;
  }

  public boolean isMultipleChoice() {
    return myIsMultipleChoice;
  }

  public void setMultipleChoice(boolean multipleChoice) {
    myIsMultipleChoice = multipleChoice;
  }

  public List<String> getChoiceVariants() {
    return myChoiceVariants;
  }

  public void setChoiceVariants(List<String> choiceVariants) {
    myChoiceVariants = choiceVariants;
  }

  public ChoiceTask(@NotNull final String name) {
    super(name);
  }

  @Override
  public String getTaskType() {
    return "choice";
  }
}

File: educational-core/src/com/jetbrains/edu/learning/courseFormat/tasks/ChoiceTask.kt
package com.jetbrains.edu.learning.courseFormat.tasks

class ChoiceTask : Task {

  var choiceOptions = listOf<ChoiceOption>()
  var isMultipleChoice: Boolean = false
  var selectedVariants = mutableListOf<Int>()

  //used for deserialization
  @Suppress("unused")
  constructor()

  constructor(name: String) : super(name)

  override fun getItemType(): String = "choice"

  class ChoiceOption {
    var text: String = ""
    var status: OptionStatus = OptionStatus.UNKNOWN

    //used for deserialization
    @Suppress("unused")
    private constructor()

    constructor(text: String) {
      this.text = text
    }

    constructor(text: String, status: OptionStatus) {
      this.text = text
      this.status = status
    }
  }

  /**
   * Choice tasks created on Stepik can't be checked locally, so they have UNKNOWN status
   */
  enum class OptionStatus {
    CORRECT, INCORRECT, UNKNOWN
  }
}
